src/components/ProtectedRoute.test.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import { initializeTestApp } from '../lib/testing';
+import ProtectedRoute from './ProtectedRoute';
+
+window.MutationObserver = require('mutation-observer');
+const component = () => <h1>Protected Header</h1>;
+
+function renderAtRoute(route) {
+  return render(
+    <MemoryRouter initialEntries={[route]}>
+      <ProtectedRoute path={route} component={component} />
+    </MemoryRouter>
+  );
+}
+
+const app = initializeTestApp({
+  projectId: 'protected-route-test',
+  apiKey: 'fake-api-key',
+  auth: { uid: '123', email: 'michaelmeyers@gmail.com' },
+});
+
+afterAll(() => app.delete());
+
+test('athenticated user can accesss page', () => {
+  renderAtRoute('/authenticated');
+
+  expect(screen.getByText('Protected Header')).toBeInTheDocument();
+});
+
+test("unathenticated user can't access page", () => {
+  app.auth().signOut();
```

Comment:
Hmm it might be the fact **ProtectedRoute** uses `firebase.auth()`. In #698 I changed `Scholarships.ts` to use `firebase.app().auth()` instead of `firebase.auth()`:
https://github.com/beyondhb1079/s4us/blob/eb8ba2ffe3a1b6de8f32c9135a39c1164c024a3b/src/models/Scholarships.ts#L12

I think I had to do that because I mocked the test app's auth function:
https://github.com/beyondhb1079/s4us/blob/eb8ba2ffe3a1b6de8f32c9135a39c1164c024a3b/src/lib/testing.ts#L45

Maybe adding a `firebase.auth = app.auth` statement before `return app` in the `initializeTestApp` method would fix this? 🤔

Reply:
hmm I tried it but it throws an error
<img width="1436" alt="Screen Shot 2021-10-31 at 11 09 37 AM" src="https://user-images.githubusercontent.com/40483569/139596355-f038dbb6-7a05-487b-b502-990d24c377fd.png">